### DISCUSSION TITLE
Is it necessary to sacrifice support for Spring versions prior to 5.3 ?

### DISCUSSION THREAD

**Discussion by robinroos** [glowroot/glowroot#972] [General]
Regarding https://github.com/glowroot/glowroot/pull/950 and the issue which it addresses https://github.com/glowroot/glowroot/issues/949, is it perhaps necessary for Glowroot to drop support for Spring versions prior to 5.3, i.e. to drop support for Spring versions which use PatternsCondition.

If it is possible simultaneously to support both invocation styles then that would be great.  5.3 is only a little over a year old, but things tend to move quite rapidly in the Spring ecosystem so that is alreay quite an "old" version.

I hope to see a fix for https://github.com/glowroot/glowroot/issues/949 (ideally one deployable via Helm) so I can determine whether a separate fix is necessary for https://github.com/glowroot/glowroot/issues/945.

**Comment by nowheresly** [comment]
Hi @robinroos ,
I share the same feeling. The latest supported versions of spring are detailed [here](https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Versions) and it seems reasonable to state that spring 5.3 is minimum required version.
